client/src/assets/player/peertube-videojs-plugin.ts

```diff
@@ -288,7 +291,8 @@ class PeerTubePlugin extends Plugin {
         renderVideo(torrent.files[ 0 ], this.playerElement, renderVideoOptions, (err, renderer) => {
           this.renderer = renderer
 
-          if (err) return this.fallbackToHttp(done)
+          console.log('value this.playerRefusedP2P', this.playerRefusedP2P)
+          if (err || this.playerRefusedP2P) return this.fallbackToHttp(done)
```

Comment:
It's not here where you should fallback to http. Use it [here](https://github.com/Chocobozzz/PeerTube/blob/develop/client/src/assets/player/peertube-videojs-plugin.ts#L436)

Reply:
@Chocobozzz placing fallback anywhere in `initializePlayer()` doesn't work.